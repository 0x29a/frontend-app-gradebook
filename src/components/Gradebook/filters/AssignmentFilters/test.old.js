import React from 'react';
import PropTypes from 'prop-types';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  Assignments,
  mapStateToProps,
  mapDispatchToProps
} from './Assignments';

jest.mock('@edx/paragon', () => ({
  Button: 'Button',
  Collapsible: 'Collapsible',
  Form: 'Form',
}));

describe("Assignments", () => {
  const props = {
    assignmentGradeMin: '0',
    assignmentGradeMax: '100',
    courseId: '12345',
    setAssignmentGradeMin: jest.fn().mockName("setAssignmentGradeMin"),
    setAssignmentGradeMax: jest.fn().mockName("setAssignmentGradeMax"),
    updateQueryParams: jest.fn().mockName("updateQueryParams"),

    assignmentTypes: ['assgn1', 'assgn2', 'assgn3'],
    assignmentFilterOptions: [
      { label: 'assgn_type1', subsectionLabel: 'subLabel1' },
      { label: 'assgn_type2', subsectionLabel: 'subLabel2' },
    ],
    filterAssignmentType: jest.fn().mockName('filterAssignmentType'),
    // TODO
    getUserGrades: jest.fn().mockName('getUserGrades').mockReturnValue(),
    selectedAssignmentType: 'assgnFilterLabel1',
    selectedAssignment: 'assgn1',
    selectedCohort: 'a cohort',
    selectedTrack: 'a track',
    updateGradesIfAssignmentGradeFiltersSet: jest.fn().mockName(
      'updateGradesIfAssignmentGradeFiltersSet',
    ),
    updateAssignmentFilter: jest.fn().mockName('updateAssignmentFilter'),
    updateAssignmentLimits: jest.fn().mockName('updateAssignmentLimits'),
  };

  describe("Component", () => {
    describe("snapshots", () => {
      const methods = [
        "handleAssignmentFilterChange",
        "handleAssignmentTypeFilterChange",
        "handleSubmitAssignmentGrade",
        "handleSetAssignmentGradeMin",
        "handleSetAssignmentGradeMax",
      ];
      let el;
      const mockHandlers = (el) => {
        const mockMethod = 
        methods.map((name) => {
          el.instance()[name] = jest.fn().mockName(name);
          return el.instance()[name];
        });
      };
      test("basic snapshot: allEnabled", () => {
        const el = shallow(<Assignments {...props} />);
        mockHandlers(el);
        expect(el.instance().render()).toMatchSnapshot();
      });

    });
  });
  describe("mapStateToProps", () => {

  });
  describe("mapDispatchToProps", () => {

  });
});
