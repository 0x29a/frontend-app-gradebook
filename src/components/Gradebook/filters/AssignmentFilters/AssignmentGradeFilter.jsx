/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Button } from '@edx/paragon';

import * as gradesActions from 'data/actions/grades';
import * as filterActions from 'data/actions/filters';

import PercentGroup from './PercentGroup';

export class AssignmentGradeFilter extends React.Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleSetMax = this.handleSetMax.bind(this);
    this.handleSetMin = this.handleSetMin.bind(this);
  };

  handleSubmit(event) {
    event.preventDefault();
    const {
      assignmentGradeMin,
      assignmentGradeMax,
    } = this.props;

    this.props.updateAssignmentLimits(
      assignmentGradeMin,
      assignmentGradeMax
    );
    this.props.getUserGrades(
      this.props.courseId,
      this.props.selectedCohort,
      this.props.selectedTrack,
      this.props.selectedAssignmentType,
    );
    this.props.updateQueryParams({
      assignmentGradeMin,
      assignmentGradeMax
    });
  }

  handleSetMax(event) {
    this.props.setAssignmentGradeMax(event.target.value);
  }

  handleSetMin(event) {
    this.props.setAssignmentGradeMin(event.target.value);
  }

  render() {
    return (
      <form
        className="grade-filter-inputs"
        onSubmit={this.handleSubmit}
      >
        <PercentGroup
          id="assignmentGradeMin"
          label="Min Grade"
          value={this.props.assignmentGradeMin}
          disabled={!this.props.selectedAssignment}
          onChange={this.handleSetMin}
        />
        <PercentGroup
          id="assignmentGradeMax"
          label="Max Grade"
          value={this.props.assignmentGradeMax}
          disabled={!this.props.selectedAssignment}
          onChange={this.handleSetMax}
        />
        <div className="grade-filter-action">
          <Button
            type="submit"
            variant="outline-secondary"
            name="assignmentGradeMinMax"
            disabled={!this.props.selectedAssignment}
          >
            Apply
          </Button>
        </div>
      </form>
    );
  }
}

AssignmentGradeFilter.defaultProps = {
  selectedAssignment: '',
  selectedAssignmentType: '',
  selectedCohort: null,
  selectedTrack: null,
};

AssignmentGradeFilter.propTypes = {
  assignmentGradeMin: PropTypes.string.isRequired,
  assignmentGradeMax: PropTypes.string.isRequired,
  courseId: PropTypes.string.isRequired,
  setAssignmentGradeMin: PropTypes.func.isRequired,
  setAssignmentGradeMax: PropTypes.func.isRequired,
  updateQueryParams: PropTypes.func.isRequired,

  // redux
  getUserGrades: PropTypes.func.isRequired,
  selectedAssignmentType: PropTypes.string,
  selectedAssignment: PropTypes.string,
  selectedCohort: PropTypes.string,
  selectedTrack: PropTypes.string,
  updateAssignmentLimits: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => ({
  selectedAssignment: (state.filters.assignment || {}).label,
  selectedAssignmentType: state.filters.assignmentType,
  selectedCohort: state.filters.cohort,
  selectedTrack: state.filters.track,
});

export const mapDispatchToProps = {
  getUserGrades: gradesActions.fetchGrades,
  updateAssignmentLimits: filterActions.updateAssignmentLimits,
};

export default connect(mapStateToProps, mapDispatchToProps)(AssignmentGradeFilter);
