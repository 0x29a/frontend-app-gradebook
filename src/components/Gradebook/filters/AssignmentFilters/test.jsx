/* eslint-disable import/no-named-as-default */

import React from 'react';
import { shallow } from 'enzyme';

import AssignmentFilters from '.';

jest.mock('@edx/paragon', () => ({
  Collapsible: 'Collapsible',
}));

describe('Assignments', () => {
  const props = {
    assignmentGradeMin: '5',
    assignmentGradeMax: '92',
    courseId: '12345',
    setAssignmentGradeMin: jest.fn().mockName('setAssignmentGradeMin'),
    setAssignmentGradeMax: jest.fn().mockName('setAssignmentGradeMax'),
    updateQueryParams: jest.fn().mockName('updateQueryParams'),
  };

  describe('Component', () => {
    describe('snapshots', () => {
      test('basic snapshot', () => {
        const el = shallow(<AssignmentFilters {...props} />);
        expect(el).toMatchSnapshot();
      });
    });
  });
  describe('mapStateToProps', () => {

  });
  describe('mapDispatchToProps', () => {

  });
});
