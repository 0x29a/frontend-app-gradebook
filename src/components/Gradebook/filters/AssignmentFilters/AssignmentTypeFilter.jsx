/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { selectableAssignmentLabels } from 'data/selectors/filters';
import * as gradesActions from 'data/actions/grades';

import SelectGroup from './SelectGroup';

export class AssignmentTypeFilter extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  };

  handleChange(event) {
    const assignmentType = event.target.value;
    this.props.filterAssignmentType(assignmentType);
    this.props.updateQueryParams({ assignmentType });
  }

  get options() {
    const mapper = (entry) => (
      <option key={entry} value={entry}>{entry}</option>
    );
    return [
      <option key="0" value="">All</option>,
      ...this.props.assignmentTypes.map(mapper),
    ];
  }

  render() {
    return (
      <SelectGroup
        id="assignment-types"
        label="Assignment Types"
        value={this.props.selectedAssignmentType}
        onChange={this.handleChange}
        disabled={this.props.assignmentFilterOptions.length === 0}
        options={this.options}
      />
    );
  }
}

AssignmentTypeFilter.defaultProps = {
  assignmentTypes: [],
  assignmentFilterOptions: [],
  selectedAssignmentType: '',
};

AssignmentTypeFilter.propTypes = {
  updateQueryParams: PropTypes.func.isRequired,

  // redux
  assignmentTypes: PropTypes.arrayOf(PropTypes.string),
  assignmentFilterOptions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    subsectionLabel: PropTypes.string,
  })),
  filterAssignmentType: PropTypes.func.isRequired,
  selectedAssignmentType: PropTypes.string,
};

export const mapStateToProps = (state) => ({
  assignmentTypes: state.assignmentTypes.results,
  assignmentFilterOptions: selectableAssignmentLabels(state),
  selectedAssignmentType: state.filters.assignmentType,
});

export const mapDispatchToProps = {
  filterAssignmentType: gradesActions.filterAssignmentType,
};

export default connect(mapStateToProps, mapDispatchToProps)(AssignmentTypeFilter);
