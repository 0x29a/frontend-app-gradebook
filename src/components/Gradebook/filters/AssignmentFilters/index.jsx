/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

import { Collapsible } from '@edx/paragon';

import AssignmentTypeFilter from './AssignmentTypeFilter';
import AssignmentFilter from './AssignmentFilter';
import AssignmentGradeFilter from './AssignmentGradeFilter';

export const AssignmentFilters = ({
  courseId,
  assignmentGradeMax,
  assignmentGradeMin,
  setAssignmentGradeMax,
  setAssignmentGradeMin,
  updateQueryParams,
}) => {
  return (
    <Collapsible title="Assignments" defaultOpen className="filter-group mb-3">
      <div>
        <AssignmentTypeFilter
          updateQueryParams={updateQueryParams}
        />
        <AssignmentFilter
          courseId={courseId}
          updateQueryParams={updateQueryParams}
        />
        <AssignmentGradeFilter
          courseId={courseId}
          assignmentGradeMin={assignmentGradeMin}
          assignmentGradeMax={assignmentGradeMax}
          setAssignmentGradeMin={setAssignmentGradeMin}
          setAssignmentGradeMax={setAssignmentGradeMax}
        />
      </div>
    </Collapsible>
  );
}

AssignmentFilters.propTypes = {
  assignmentGradeMin: PropTypes.string.isRequired,
  assignmentGradeMax: PropTypes.string.isRequired,
  courseId: PropTypes.string.isRequired,
  setAssignmentGradeMin: PropTypes.func.isRequired,
  setAssignmentGradeMax: PropTypes.func.isRequired,
  updateQueryParams: PropTypes.func.isRequired,
};

export default AssignmentFilters;
