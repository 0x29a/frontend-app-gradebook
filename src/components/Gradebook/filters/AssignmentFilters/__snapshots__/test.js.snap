// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments Component snapshots basic snapshot 1`] = `
<Collapsible
  className="filter-group mb-3"
  defaultOpen={true}
  title="Assignments"
>
  <div>
    <Connect(AssignmentTypeFilter)
      updateQueryParams={[MockFunction updateQueryParams]}
    />
    <Connect(AssignmentFilter)
      courseId="12345"
      updateQueryParams={[MockFunction updateQueryParams]}
    />
    <Connect(AssignmentGradeFilter)
      assignmentGradeMax="92"
      assignmentGradeMin="5"
      courseId="12345"
      setAssignmentGradeMax={[MockFunction setAssignmentGradeMax]}
      setAssignmentGradeMin={[MockFunction setAssignmentGradeMin]}
    />
  </div>
</Collapsible>
`;
